### PR TITLE
Add email and description for guest and for customer with is_vault_en…

### PR DESCRIPTION
**Pull request by SaftaArtem**
- In `view/frontend/web/js/view/payment/adapter.js` I get  enable_vault checkbox value to transfer in Payment Intent Create Controller
- In Payment Intent Create controller i handle enable_vault value and set email and description for guest and customers logged in
- In `Gateway/Helper/SubjectReader.php` I created method for getting from payment order id from payment and format it.
- In `Gateway/Request/PaymentDataBuilder.php` I set order id to description in case when payment intent does not create
- In `Model/Adapter/StripeAdapter.php` I update description in case: extension use existing payment intent

### CODE REVIEW

**Comment by SaftaArtem** [comment]
- In `view/frontend/web/js/view/payment/adapter.js` I get  enable_vault checkbox value to transfer in Payment Intent Create Controller
- In Payment Intent Create controller i handle enable_vault value and set email and description for guest and customers logged in
- In `Gateway/Helper/SubjectReader.php` I created method for getting from payment order id and format it.

File: view/frontend/web/js/view/payment/adapter.js [hunk 1]
```diff
@@ -122,6 +122,10 @@ define([
         createPaymentIntent: function () {
             var self = this,
                 dfd = $.Deferred();
+            if ($("#tnw_stripe_enable_vault").length) {
+                console.log(customerData);
```
Comment (vsevolod-tarasov) [inline view/frontend/web/js/view/payment/adapter.js:126]: we should not output customerData to console

File: Gateway/Helper/SubjectReader.php [hunk 2]
```diff
@@ -68,4 +70,10 @@ public function readCustomerId(array $subject)
 
         return (int) $subject['customer_id'];
     }
+
+    public function getOrderIncrementId(\Magento\Sales\Api\Data\OrderPaymentInterface $payment)
+    {
+        $incrementId = $payment->getOrder()->getIncrementId();
```
Comment (vsevolod-tarasov) [inline Gateway/Helper/SubjectReader.php:76]: there is not need in variable creation

File: Gateway/Helper/SubjectReader.php [hunk 3]
```diff
@@ -68,4 +70,10 @@ public function readCustomerId(array $subject)
 
         return (int) $subject['customer_id'];
     }
+
+    public function getOrderIncrementId(\Magento\Sales\Api\Data\OrderPaymentInterface $payment)
```
Comment (vsevolod-tarasov) [inline Gateway/Helper/SubjectReader.php:74]: should be placed into usages

File: Gateway/Helper/SubjectReader.php (hunk 2 above)
Comment (idenys150) [reply Gateway/Helper/SubjectReader.php]: Thanks, fixed

File: Gateway/Helper/SubjectReader.php (hunk 3 above)
Comment (idenys150) [reply Gateway/Helper/SubjectReader.php]: Thanks - done

File: Model/Adapter/StripeAdapter.php [hunk 4]
```diff
@@ -90,6 +90,7 @@ public function sale(array $attributes)
             $pi = PaymentIntent::create($attributes);
         } else {
             $pi = PaymentIntent::retrieve($attributes['pi']);
+            if (isset($attributes['description'])) $pi->update($attributes['pi'], ['description' => $attributes['description']]);
```
[the remark this answers is not in the document]
Comment (idenys150) [reply Model/Adapter/StripeAdapter.php]: Format has been adjusted

File: view/frontend/web/js/view/payment/adapter.js (hunk 1 above)
Comment (idenys150) [reply view/frontend/web/js/view/payment/adapter.js]: Thanks, removed

File: Console/Command/GuestOrder.php [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+use Magento\Framework\App\Area;
+use Magento\Framework\App\State;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use TNW\Stripe\Model\Order\Guest;
+/**
+ * Class GuestOrder
+ * @package TNW\Stripe\Console\Command
```
Comment (vsevolod-tarasov) [inline Console/Command/GuestOrder.php:12]: package directive is redundant - see magento 2 coding standard or use their codesniffer to check

File: Console/Command/GuestOrder.php [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+        parent::configure();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $this->appState->emulateAreaCode(Area::AREA_FRONTEND, [$this, 'export'], [$input, $output]);
+
```
Comment (vsevolod-tarasov) [inline Console/Command/GuestOrder.php:59]: redundant line

File: Controller/PaymentIntent/Create.php [hunk 7]
```diff
@@ -7,10 +7,14 @@
...
 use TNW\Stripe\Model\Adapter\StripeAdapterFactory;
+use TNW\Stripe\Helper\Customer as CustomerHelper;
 use Magento\Framework\Controller\ResultFactory;
+use Magento\Customer\Model\Session;
+use Magento\Checkout\Model\Session as CheckoutSession;
 
 /**
  * Class Create
+ *
  * @package TNW\Stripe\Controller\Paymentintent
```
Comment (vsevolod-tarasov) [inline Controller/PaymentIntent/Create.php:18]: package is redundant

File: Controller/PaymentIntent/Create.php [hunk 8]
```diff
@@ -65,13 +89,30 @@ public function execute()
...
+            $attributes = [
+                'payment_method' => $payment->id,
+                'metadata' => ['site' => $this->_url->getBaseUrl()]
+            ];
+
+
+            // for login customers while enable vault in configuration and vault not checked in frontend case
+            $email = $this->checkoutSession->getQuote()->getBillingAddress()->getEmail();
+            $isLoggedIn = $this->session->isLoggedIn();
+            if (($isLoggedIn)) {
```
Comment (vsevolod-tarasov) [inline Controller/PaymentIntent/Create.php:106]: redundant ()

File: Controller/PaymentIntent/Create.php [hunk 9]
```diff
@@ -65,13 +89,30 @@ public function execute()
...
+                $attributes['email'] = $email;
+            } elseif (!$isLoggedIn) {
```
Comment (vsevolod-tarasov) [inline Controller/PaymentIntent/Create.php:108]: else here will be enough

File: Gateway/Helper/SubjectReader.php [hunk 10]
```diff
@@ -16,13 +16,20 @@
 namespace TNW\Stripe\Gateway\Helper;
 
 use Magento\Payment\Gateway\Helper;
+use Magento\Sales\Api\Data\OrderPaymentInterface;
 
+/**
+ * Class SubjectReader
+ * @package TNW\Stripe\Gateway\Helper
```
Comment (vsevolod-tarasov) [inline Gateway/Helper/SubjectReader.php:23]: same for class header

File: Gateway/Helper/SubjectReader.php [hunk 11]
```diff
@@ -68,4 +75,9 @@ public function readCustomerId(array $subject)
 
         return (int) $subject['customer_id'];
     }
+
+    public function getOrderIncrementId(OrderPaymentInterface $payment)
```
Comment (vsevolod-tarasov) [inline Gateway/Helper/SubjectReader.php:79]: missing comment for method

File: Helper/Customer.php [hunk 12]
```diff
@@ -0,0 +1,45 @@
+<?php
+namespace TNW\Stripe\Helper;
+
+use Magento\Customer\Api\CustomerRepositoryInterface;
+/**
```
Comment (vsevolod-tarasov) [inline Helper/Customer.php:5]: missing line

File: Helper/Customer.php [hunk 13]
```diff
@@ -0,0 +1,45 @@
...
+ * Class Customer
+ * @package TNW\Stripe\Helper
```
Comment (vsevolod-tarasov) [inline Helper/Customer.php:7]: invalid class doc

File: Helper/Customer.php [hunk 14]
```diff
@@ -0,0 +1,45 @@
...
+     * @throws \Magento\Framework\Exception\InputException
+     * @throws \Magento\Framework\Exception\LocalizedException
+     * @throws \Magento\Framework\Exception\NoSuchEntityException
+     * @throws \Magento\Framework\Exception\State\InputMismatchException
+     */
+    public function updateCustomerStripeId($email, $cs)
+    {
+        try {
+            $customer = $this->customerRepository->get($email);
+            //$customer->setData('stripe_id', $cs);
```
Comment (vsevolod-tarasov) [inline Helper/Customer.php:38]: commented code should not be commited

File: Model/Order/Guest.php [hunk 15]
```diff
@@ -0,0 +1,117 @@
...
+
+namespace TNW\Stripe\Model\Order;
+
+use Magento\Framework\Controller\ResultFactory;
+use Magento\Sales\Model\ResourceModel\Order\CollectionFactory;
+use Magento\Sales\Model\ResourceModel\Order\Collection;
+use TNW\Stripe\Helper\Customer as CustomerHelper;
+use TNW\Stripe\Model\Adapter\StripeAdapterFactory;
+use TNW\Stripe\Helper\Payment\Formatter;
+/**
```
Comment (vsevolod-tarasov) [inline Model/Order/Guest.php:11]: missing line and invalid class doc

File: Model/Order/Guest.php [hunk 16]
```diff
@@ -0,0 +1,117 @@
...
+    public function getGuestOrders()
+    {
+        /**
+         * @var Collection $orderCollecion
+         */
+        $orderCollecion = $this->orderCollectionFactory
+            ->create()
+            ->addFieldToSelect('*');
+
+        $orderCollecion
```
Comment (vsevolod-tarasov) [inline Model/Order/Guest.php:62]: put those addAttributeToFilter into above construction

Comment (vsevolod-tarasov) [reply Model/Order/Guest.php:62]: there is no need in addition variable here just:
`


return $this->orderCollectionFactory
->create()
 ->addFieldToSelect('*')
->addAttributeToFilter('customer_is_guest', ['eq' => 1])
->addAttributeToFilter('guest_order_exported', ['null' => null]);
`

File: Model/Order/Guest.php [hunk 17]
```diff
@@ -0,0 +1,117 @@
...
+
+    /**
+     * Export Guest Orders
+     *
+     * @return array
+     */
+    public function exportGuestOrders()
+    {
+        $response = [];
+        $orderCollection = $this->getGuestOrders();
```
Comment (vsevolod-tarasov) [inline Model/Order/Guest.php:76]: no need it additional var here

File: Model/Order/Guest.php [hunk 18]
```diff
@@ -0,0 +1,117 @@
...
+            $paymentIntent = $stripeAdapter->retrievePaymentIntent($transactionId);
+
+            $stripeAdapter->retrievePaymentIntent($paymentIntent->id);
+            $cid = $paymentIntent->customer;
+            $arrayParams = [
+                'email' => $guestOrder->getCustomerEmail(),
+            ];
+            $stripeAdapter->updateCustomer($cid, $arrayParams);
+            $guestOrder->setGuestOrderExported(1)->save();
+
```
Comment (vsevolod-tarasov) [inline Model/Order/Guest.php:111]: redundant line

File: Setup/UpgradeData.php [hunk 19]
```diff
@@ -50,4 +111,66 @@ protected function version_2_1_7(
...
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function version_2_3_2(
+        ModuleDataSetupInterface $setup
+    ) {
+        $eavSetup = $this->eavSetupFactory->create(['setup' => $setup]);
+        $eavSetup->removeAttribute(
+            \Magento\Customer\Model\Customer::ENTITY,
```
Comment (vsevolod-tarasov) [inline Setup/UpgradeData.php:123]: we already have the \Magento\Customer\Model\Customer in usages there is no need in full class path here

File: Setup/UpgradeData.php [hunk 20]
```diff
@@ -50,4 +111,66 @@ protected function version_2_1_7(
...
+            'input' => 'text',
+            'required' => false,
+            'visible' => true,
+            'user_defined' => true,
+            'sort_order' => 1000,
+            'position' => 1000,
+            'system' => 0,
+        ]);
+        //add attribute to attribute set
+        $attribute = $customerSetup->getEavConfig()->getAttribute(Customer::ENTITY, 'stripe_id')
```
Comment (vsevolod-tarasov) [inline Setup/UpgradeData.php:148]: we can remove this variable and call save after addData with chain-call

File: Setup/UpgradeData.php [hunk 21]
```diff
@@ -50,4 +111,66 @@ protected function version_2_1_7(
...
+     */
+    protected function version_2_3_3($setup)
+    {
+        /** @var \Magento\Sales\Setup\SalesSetup $salesInstaller */
+        $salesInstaller = $this->salesSetupFactory->create(['resourceName' => 'sales_setup', 'setup' => $setup]);
+        $salesInstaller->addAttribute(
+            'order',
+            'guest_order_exported',
+            [
+                'type' => \Magento\Framework\DB\Ddl\Table::TYPE_TEXT,
```
Comment (vsevolod-tarasov) [inline Setup/UpgradeData.php:169]: place \Magento\Framework\DB\Ddl\Table into usages

File: Setup/UpgradeData.php [hunk 22]
```diff
@@ -30,6 +84,13 @@ public function upgrade(
             $this->version_2_1_7($context, $setup);
         }
 
+        if (version_compare($context->getVersion(), "2.3.2", "<=")) {
```
Comment (vsevolod-tarasov) [inline Setup/UpgradeData.php:87]: we added two version updates here. Starting version is 2.3.2. Check here should be <. And should be done in one version up both upgrades for version 2.3.3.
Module version should be upgraded in module.xml not to 2.3.4 but to 2.3.3

File: Console/Command/GuestOrder.php [hunk 23]
```diff
@@ -0,0 +1,76 @@
+<?php
+namespace TNW\Stripe\Console\Command;
+
+use Magento\Framework\App\Area;
+use Magento\Framework\App\State;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use TNW\Stripe\Model\Order\Guest;
+/**
```
[the remark this answers is not in the document]
Comment (diitd) [reply Console/Command/GuestOrder.php:11]: thanks - removed

File: Console/Command/GuestOrder.php (hunk 5 above)
Comment (diitd) [reply Console/Command/GuestOrder.php]: thanks - removed

File: Console/Command/GuestOrder.php (hunk 23 above)
[the remark this answers is not in the document]
Comment (idenys150) [reply Console/Command/GuestOrder.php:11]: Thanks - removed

File: Console/Command/GuestOrder.php (hunk 5 above)
Comment (idenys150) [reply Console/Command/GuestOrder.php]: Thanks - removed

File: Controller/PaymentIntent/Create.php [hunk 24]
```diff
@@ -65,13 +89,30 @@ public function execute()
...
+        $payment = $data->paymentMethod;
+        $amount = $data->amount;
         $currency = $data->currency;
 
         try {
+            $attributes = [
+                'payment_method' => $payment->id,
+                'metadata' => ['site' => $this->_url->getBaseUrl()]
+            ];
+
```
[the remark this answers is not in the document]
Comment (idenys150) [reply Controller/PaymentIntent/Create.php:101]: Thanks - removed

File: Console/Command/GuestOrder.php (hunk 6 above)
Comment (idenys150) [reply Console/Command/GuestOrder.php]: Thanks - removed

File: Console/Command/GuestOrder.php [hunk 25]
```diff
@@ -0,0 +1,76 @@
...
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure()
+    {
+        $this->setName('tnw:exportguestorders');
+        $this->setDescription('Export Guest Orders to Stripe');
+
```
[the remark this answers is not in the document]
Comment (idenys150) [reply Console/Command/GuestOrder.php]: Thanks - removed

File: Setup/UpgradeData.php (hunk 19 above)
Comment (idenys150) [reply Setup/UpgradeData.php]: Thanks - done

File: Setup/UpgradeData.php (hunk 20 above)
Comment (idenys150) [reply Setup/UpgradeData.php]: Thanks - done

File: Setup/UpgradeData.php (hunk 21 above)
Comment (idenys150) [reply Setup/UpgradeData.php]: Thanks - done

File: Setup/UpgradeData.php (hunk 22 above)
Comment (idenys150) [reply Setup/UpgradeData.php]: Thanks - done

File: Console/Command/GuestOrder.php [hunk 26]
```diff
@@ -0,0 +1,76 @@
+<?php
+namespace TNW\Stripe\Console\Command;
```
[the remark this answers is not in the document]
Comment (idenys150) [reply Console/Command/GuestOrder.php:2]: done

File: Console/Command/GuestOrder.php [hunk 27]
```diff
@@ -0,0 +1,76 @@
...
+
+use Magento\Framework\App\Area;
+use Magento\Framework\App\State;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use TNW\Stripe\Model\Order\Guest;
+/**
+ * Class GuestOrder
```
[the remark this answers is not in the document]
Comment (idenys150) [reply Console/Command/GuestOrder.php:12]: done

File: Controller/PaymentIntent/Create.php [hunk 28]
```diff
@@ -7,10 +7,14 @@
 use TNW\Stripe\Helper\Payment\Formatter;
 use TNW\Stripe\Gateway\Config\Config;
 use TNW\Stripe\Model\Adapter\StripeAdapterFactory;
+use TNW\Stripe\Helper\Customer as CustomerHelper;
 use Magento\Framework\Controller\ResultFactory;
+use Magento\Customer\Model\Session;
+use Magento\Checkout\Model\Session as CheckoutSession;
 
 /**
  * Class Create
```
[the remark this answers is not in the document]
Comment (idenys150) [reply Controller/PaymentIntent/Create.php:16]: done

File: Controller/PaymentIntent/Create.php (hunk 7 above)
Comment (idenys150) [reply Controller/PaymentIntent/Create.php:18]: done

File: Controller/PaymentIntent/Create.php (hunk 8 above)
Comment (idenys150) [reply Controller/PaymentIntent/Create.php:105]: done

File: Controller/PaymentIntent/Create.php (hunk 9 above)
Comment (idenys150) [reply Controller/PaymentIntent/Create.php:107]: done

File: Gateway/Helper/SubjectReader.php (hunk 10 above)
Comment (idenys150) [reply Gateway/Helper/SubjectReader.php:23]: done

File: Gateway/Helper/SubjectReader.php (hunk 11 above)
Comment (idenys150) [reply Gateway/Helper/SubjectReader.php:79]: done

File: Helper/Customer.php (hunk 12 above)
Comment (idenys150) [reply Helper/Customer.php:5]: added

**Review by vsevolod-tarasov — APPROVED** [review]

**Comment by alexander-babenkocommerce** [comment]
Not relevant

**Pull request closed without merging**